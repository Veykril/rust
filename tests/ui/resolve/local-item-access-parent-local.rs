#![allow(unused_variables, dead_code)]

fn f2() {
    let foo = 0;
    fn f() {
        foo;
        //~^ ERROR can't capture dynamic
    }
    const C: () = {
        foo;
        //~^ ERROR attempt to
    };
    static S: () = {
        foo;
        //~^ ERROR attempt to
    };
}

fn main() {}
