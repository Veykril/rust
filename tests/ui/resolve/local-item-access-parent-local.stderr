error[E0434]: can't capture dynamic environment in a fn item
  --> $DIR/local-item-access-parent-local.rs:6:9
   |
LL |         foo;
   |         ^^^
   |
   = help: use the `|| { ... }` closure form instead

error[E0435]: attempt to use a non-constant value in a constant
  --> $DIR/local-item-access-parent-local.rs:10:9
   |
LL |     const C: () = {
   |     ------- help: consider using `let` instead of `const`: `let C`
LL |         foo;
   |         ^^^ non-constant value

error[E0435]: attempt to use a non-constant value in a constant
  --> $DIR/local-item-access-parent-local.rs:14:9
   |
LL |     static S: () = {
   |     -------- help: consider using `let` instead of `static`: `let S`
LL |         foo;
   |         ^^^ non-constant value

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0434, E0435.
For more information about an error, try `rustc --explain E0434`.
